### PR TITLE
Add log to happy path

### PR DESCRIPTION
**Pull request by bohuini**
**Reason for Change**:
DNC requiring to refresh CNS NC associations is not expected to happen too often, but when it does, we need to log all pertinent details. Added the log in CNS should to show refresh detected and initiated by DNC.

**Issue Fixed**:

**Requirements**:

- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests

**Notes**:

### CODE REVIEW

**Comment by rbtr** [comment]
@bohuini pls read this https://github.com/Azure/azure-container-networking/pull/1921#issuecomment-1530069378

File: cns/restserver/api.go [hunk 1]
```diff
@@ -919,11 +919,15 @@ func (service *HTTPRestService) getNetworkContainerByOrchestratorContext(w http.
 // If received  "GET": Return all NCs in CNS's state file to DNC in order to check if NC refresh is needed
 // If received "POST": Store all the NCs (from the request body that client sent) into CNS's state file
 func (service *HTTPRestService) getOrRefreshNetworkContainers(w http.ResponseWriter, r *http.Request) {
+	logger.Printf("[Azure CNS] NC refresh is initiated by DNC. The state file in CNS will get updated if it is lost.")
```
Comment (rbtr) [inline cns/restserver/api.go:922]: unnecessary to add an informational log here if all branches of the switch have a log

File: cns/restserver/api.go [hunk 2]
```diff
@@ -919,11 +919,15 @@ func (service *HTTPRestService) getNetworkContainerByOrchestratorContext(w http.
...
+
 	switch r.Method {
 	case http.MethodGet:
+		logger.Printf("[Azure CNS] DNC is attempting to check if the state file in CNS is intact.")
```
Comment (rbtr) [inline cns/restserver/api.go:926]: or, `GET from getOrRefreshNetworkContainers`

File: cns/restserver/api.go [hunk 3]
```diff
@@ -919,11 +919,15 @@ func (service *HTTPRestService) getNetworkContainerByOrchestratorContext(w http.
...
 		service.handleGetNetworkContainers(w)
 		return
 	case http.MethodPost:
+		logger.Printf("[Azure CNS] The state file in CNS is lost. CNS will store all the NCs sent from DNC into state file.")
```
Comment (rbtr) [inline cns/restserver/api.go:930]: or, `POST to getOrRefreshNetworkContainers`

**Review by rbtr — CHANGES_REQUESTED** [review]
the log should not speculate on the _intent_ of what is happening, the log should describe what is happening concisely and in enough detail to be useful during debugging

File: cns/restserver/api.go (hunk 2 above)
Comment (rbtr) [reply cns/restserver/api.go:926]: you don't have to use my example, but it provides just as much info as yours without the potential of lying to me if that's not why this method was called

**Comment by timraymond** [comment]
I'm with @rbtr here... I think concise logs are better because they tend to tell fewer lies about what's going on as things change over time. Log just the facts, given what's known at the time the log is generated.  Given what I see, it would likely be better to have a middleware that logs requests (path, method, some request ID), and responses (HTTP response code, end-to-end time, the same ID for correlation). All of those should be structured using zap. Something like:

```go
func logHTTP(next http.Handler) http.Handler {
	return http.HandlerFunc(func(rw http.ResponseWriter, r *http.Request) {
		logger.Info("request", zap.String("uri", r.URL.RequestURI()), zap.String("method", r.Method))
		// need to wrap the ResponseWriter in something that can capture the response code
		wrw := WrappedResponseWriter{rw} // this doesn't exist and needs to be written
		next(wrw, r)
		logger.Info("response", zap.Int("code", wrw.ResponseCode)) // etc.
	})
}

**Comment by bohuini** [comment]
@rbtr Revised the log and rebased branch. Please review again when you get a chance. Thanks!

**Comment by bohuini** [comment]
@timraymond That's a good point and I do agree. But for this time I am just making it simple for integration testing purposes.

File: cns/restserver/api.go (hunk 1 above)
Comment (bohuini) [reply cns/restserver/api.go]: Removed.

File: cns/restserver/api.go (hunk 2 above)
Comment (bohuini) [reply cns/restserver/api.go]: Make sense. Addressed.

File: cns/restserver/api.go (hunk 3 above)
Comment (bohuini) [reply cns/restserver/api.go]: Addressed.

**Comment by timraymond** [comment]
LGTM 👍 Awaiting @rbtr to hit the approval button.

**Review by rbtr — APPROVED** [review]

**Pull request merged**